benchmarks/util/gogoproto_generator.h

```diff
@@ -0,0 +1,98 @@
+// TODO: Insert description here. (generated by yilunchong)
+
+#ifndef PROTOBUF_BENCHMARKS_UTIL_GOGOPROTO_GENERATOR_H_
+#define PROTOBUF_BENCHMARKS_UTIL_GOGOPROTO_GENERATOR_H_
+
+#include "google/protobuf/compiler/code_generator.h"
+#include "google/protobuf/io/zero_copy_stream.h"
+#include "google/protobuf/io/printer.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/descriptor.pb.h"
+#include "schema_proto2_to_proto3_util.h"
+
+#include <fstream>
+
+using google::protobuf::FileDescriptorProto;
+using google::protobuf::FileDescriptor;
+using google::protobuf::DescriptorPool;
+using google::protobuf::io::Printer;
+using google::protobuf::util::SchemaGroupStripper;
+using google::protobuf::util::SchemaAddZeroEnumValue;
+
+namespace google {
+namespace protobuf {
+namespace compiler {
+
+namespace {
+
+string StripProto(string filename) {
+  if (filename.substr(filename.size() - 11) == ".protodevel") {
+    // .protodevel
+    return filename.substr(0, filename.size() - 11);
+  } else {
+    // .proto
+    return filename.substr(0, filename.size() - 6);
+  }
+}
+
+DescriptorPool new_pool_;
```

Comment:
Please avoid using global objects. Why not make this a member of the GoGoProtoGenerator below?

Reply:
The Generate function need "this" to be const. I'm not sure in which way I can avoid global objects.